contracts/0.8.9/oracle/ValidatorsExitBusOracle.sol

```diff
@@ -0,0 +1,507 @@
+// SPDX-FileCopyrightText: 2023 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.9;
+
+import { SafeCast } from "@openzeppelin/contracts-v4.4/utils/math/SafeCast.sol";
+
+import { Math } from "../lib/Math.sol";
+import { UnstructuredStorage } from "../lib/UnstructuredStorage.sol";
+import { AllowanceBasedRateLimit as RateLimit } from "../lib/AllowanceBasedRateLimit.sol";
+
+import { BaseOracle } from "./BaseOracle.sol";
+
+
+contract ValidatorsExitBusOracle is BaseOracle {
+    using RateLimit for RateLimit.State;
+    using UnstructuredStorage for bytes32;
+    using SafeCast for uint256;
+
+    error AdminCannotBeZero();
+    error SenderNotAllowed();
+    error UnsupportedRequestsDataFormat(uint256 format);
+    error InvalidRequestsData();
+    error InvalidRequestsDataLength();
+    error UnexpectedRequestsDataLength();
+    error InvalidRequestsDataSortOrder();
+    error ArgumentOutOfBounds();
+
+    event DataBoundariesSet(
+        uint256 indexed refSlot,
+        uint256 maxExitRequestsPerReport,
+        uint256 maxExitRequestsListLength,
+        uint256 exitRequestsRateLimitWindowSizeSlots,
+        uint256 exitRequestsRateLimitMaxThroughputE18
+    );
+
+    event ValidatorExitRequest(
+        uint256 indexed stakingModuleId,
+        uint256 indexed nodeOperatorId,
+        uint256 indexed validatorIndex,
+        bytes validatorPubkey,
+        uint256 timestamp
+    );
+
+    event WarnDataIncompleteProcessing(
+        uint256 indexed refSlot,
+        uint256 requestsProcessed,
+        uint256 requestsCount
+    );
+
+    struct DataBoundaries {
+        uint64 maxRequestsPerReport;
+        uint64 maxRequestsListLength;
+    }
+
+    struct DataProcessingState {
+        uint256 lastProcessedItemWithoutPubkey;
+        uint64 refSlot;
+        uint64 requestsCount;
+        uint64 requestsProcessed;
+        uint16 dataFormat;
+    }
+
+    struct RequestedValidator {
```

Comment:
can we just have `uint72 index;` and store last requested + 1 number

0 - no validator war requested
1 - means 0
2- means 1
and so on....

Reply:
1-based indices work here, but why not use a struct with a flag like done now? it's arguably more readable, doesn't increase gas costs, and less prone to errors?